index/miner/miner.go

```diff
@@ -33,41 +39,51 @@ type API interface {
 	ChainNotify(context.Context) (<-chan []*types.HeadChange, error)
 	ChainHead(context.Context) (*types.TipSet, error)
 	ChainGetTipSet(context.Context, types.TipSetKey) (*types.TipSet, error)
+	ChainGetTipSetByHeight(context.Context, uint64, *types.TipSet) (*types.TipSet, error)
 	StateChangedActors(context.Context, cid.Cid, cid.Cid) (map[string]types.Actor, error)
 	StateReadState(ctx context.Context, act *types.Actor, ts *types.TipSet) (*types.ActorState, error)
+	StateMinerPeerID(ctx context.Context, m string, ts *types.TipSet) (peer.ID, error)
 }
 
 type MinerIndex struct {
 	c        API
 	ds       datastore.TxnDatastore
+	h        *fchost.FilecoinHost
+	lr       iplocation.LocationResolver
 	signaler *signaler.Signaler
 
-	info IndexInfo
+	lock  sync.Mutex
+	index Index
 
-	lock     sync.Mutex
 	ctx      context.Context
 	cancel   context.CancelFunc
 	finished chan struct{}
 	closed   bool
 }
 
-type IndexInfo struct {
+type Index struct {
 	LastUpdated uint64
-	Power       map[string]*PowerInfo
-	// ToDo: geography/region information of miner
+	Miners      map[string]Info
+}
+
+type Info struct {
+	Static Static
+	Power  Power
 }
 
-type PowerInfo struct {
-	Power      uint64
-	TotalPower uint64
+type Power struct {
+	Power    uint64
+	Relative float64
```

Comment:
`Power`: power of miner in the network
`Relative`: relative power compared to total network power.